system/jlib/jdebug.cpp

```diff
@@ -1103,8 +1105,37 @@ static unsigned evalAffinityCpus()
     return 1;
 }
 
+// Note - values are returned in Kb
+
 static void getMemUsage(unsigned &inuse,unsigned &active,unsigned &total,unsigned &swaptotal,unsigned &swapinuse) 
 {
+#ifdef __APPLE__
+    int mib [] = { CTL_HW, HW_MEMSIZE };
+    int64_t totalBytes = 0;
+    size_t length = sizeof(totalBytes);
+
+    if(-1 == sysctl(mib, 2, &totalBytes, &length, NULL, 0))
+        return;
+
+    vm_size_t pageSize;
+    if (KERN_SUCCESS != host_page_size(mach_host_self(), &pageSize))
+        return;
+    mach_msg_type_number_t count = HOST_VM_INFO_COUNT;
+    vm_statistics64_data_t vmstat;
+    if(KERN_SUCCESS != host_statistics64(mach_host_self(), HOST_VM_INFO, (host_info_t)&vmstat, &count))
+        return;
+
+    uint64_t inactiveBytes = (vmstat.inactive_count + vmstat.free_count) * pageSize;
+    uint64_t activeBytes = totalBytes - inactiveBytes;
+    uint64_t inuseBytes = (vmstat.wire_count + vmstat.active_count) * pageSize;
+
+    active = activeBytes / 1024;
+    inuse = inuseBytes / 1024;
+    total = totalBytes/1024;
+    swaptotal = 0;
```

Comment:
minor: should there be a // MORE ? - it looks like you can get someswap stats via sysctlbyname(, e.g. sysctlbyname("vm.swapusage")